### PR TITLE
dev-python/pydotplus: new package

### PR DESCRIPTION
**Pull request by TheChymera**
@gktrk we talked about migrating this to the main tree.

Package-Manager: Portage-2.3.3, Repoman-2.3.1

### CODE REVIEW

File: dev-python/pydotplus/pydotplus-2.0.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+inherit distutils-r1
+
+DESCRIPTION="Improved version of the old pydot project"
+HOMEPAGE="http://pydotplus.readthedocs.org/"
+SRC_URI="mirror://pypi/p/${PN}/${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
```
Comment (SoapGentoo) [inline dev-python/pydotplus/pydotplus-2.0.2.ebuild]: why do you need the `test` USE flag

**Comment by gktrk** [comment]
@TheChymera I have a paper deadline in 48 hours, followed up by an interstate travel. Mind if I take a look at this on Saturday?

File: dev-python/pydotplus/pydotplus-2.0.2.ebuild (hunk 1 above)
Comment (TheChymera) [inline dev-python/pydotplus/pydotplus-2.0.2.ebuild]: Ideally I would like to be able to try and run the tests of the package.

Now I notice that the python_test() phase finds no tests to run.

I tried changing the line to `${EPYTHON} -m unittest test || die` which also finds no tests. I tried to point unittest to the script directly (via `${EPYTHON} -m unittest test/pydot_unittest.py || die`) and got this http://bpaste.net/show/94697a5519f0 .

Lastly, I followed the advice from here https://wiki.gentoo.org/wiki/Project:Python/Tests and tred navigating to the workdir to run `unit2 discover`. That also found no tests to run.

Comment (TheChymera) [inline dev-python/pydotplus/pydotplus-2.0.2.ebuild]: What can I do to run the tests?

Comment (SoapGentoo) [inline dev-python/pydotplus/pydotplus-2.0.2.ebuild]: `IUSE="test"` is orthogonal to running tests. `IUSE="test"` is used for pulling in dependencies needed for running tests, not actually for enabling/disabling tests (that is done by `FEATURES="test"`, unless you use paludis)

**Comment by TheChymera** [comment]
@SoapGentoo how about actually runnign some tests? any advice, or is it an upstream issue?

**Comment by SoapGentoo** [comment]
@TheChymera 
check the upstream doc (https://github.com/carlos-jenkins/pydotplus):
Run code QA:
```
pip install tox
tox
```
Check `deps=` the `tox.ini`:

1. **flake8**
2. **pytest**
3. **pytest-cov**
4. **sphinx**
5. **sphinx_rtd_theme**

so reintroduce `IUSE=test`, and put these in a `test?` conditional in `DEPEND`

**Comment by TheChymera** [comment]
@SoapGentoo how can I test the `test` feature just for pydotplus?

```
zenbookhost ~ # FEATURES=test emerge -v pydotplus

These are the packages that would be merged, in order:

Calculating dependencies... done!


[ebuild   R    ] dev-python/pydotplus-2.0.2::neurogentoo  USE="{test*}" PYTHON_TARGETS="python2_7 python3_4 -python3_5" 0 KiB
[nomerge       ]  dev-python/sphinx-1.5.2::gentoo  USE="{test} -doc -latex -net" PYTHON_TARGETS="python2_7 python3_4 -python3_5 (-python3_6)" 
[nomerge       ]   dev-python/flake8-3.3.0::gentoo  USE="{test}" PYTHON_TARGETS="python2_7 python3_4 -pypy -pypy3 -python3_5 (-python3_6)" 
[ebuild  N     ]    dev-python/mccabe-0.6.1::gentoo  USE="{test}" PYTHON_TARGETS="python2_7 python3_4 -pypy -pypy3 -python3_5 (-python3_6)" 9 KiB
[ebuild  N     ]  dev-python/sphinx-1.5.2::gentoo  USE="{test} -doc -latex -net" PYTHON_TARGETS="python2_7 python3_4 -python3_5 (-python3_6)" 4'316 KiB
[ebuild  N     ]   dev-python/flake8-3.3.0::gentoo  USE="{test}" PYTHON_TARGETS="python2_7 python3_4 -pypy -pypy3 -python3_5 (-python3_6)" 132 KiB

Total: 4 packages (3 new, 1 reinstall), Size of downloads: 4'455 KiB

 * Error: circular dependencies:

(dev-python/mccabe-0.6.1:0/0::gentoo, ebuild scheduled for merge) depends on
 (dev-python/flake8-3.3.0:0/0::gentoo, ebuild scheduled for merge) (runtime)
  (dev-python/mccabe-0.6.1:0/0::gentoo, ebuild scheduled for merge) (buildtime)

It might be possible to break this cycle
by applying the following change:
- dev-python/flake8-3.3.0 (Change USE: -test)

Note that this change can be reverted, once the package has been installed.
```

**Comment by SoapGentoo** [comment]
@TheChymera try first installing the deps `emerge -vo1 --with-test-deps pydotplus`
Then try `FEATURES=test ebuild pydotplus-blabla.ebuild test`

**Comment by TheChymera** [comment]
@SoapGentoo did that, and the test phase looks bigger now - but I can't see any tests being actually run. Any idea what's going on here?:

```
Forcing test.
>>> Existing ${T}/environment for 'pydotplus-2.0.2' will be sourced. Run
>>> 'clean' to start with a fresh environment.
>>> Checking pydotplus-2.0.2.tar.gz's mtime...
>>> WORKDIR is up-to-date, keeping...
>>> It appears that 'pretend' has already executed for 'pydotplus-2.0.2'; skipping.
>>> Remove '/var/tmp/portage/dev-python/pydotplus-2.0.2/.pretended' to force pretend.
>>> It appears that 'setup' has already executed for 'pydotplus-2.0.2'; skipping.
>>> Remove '/var/tmp/portage/dev-python/pydotplus-2.0.2/.setuped' to force setup.
>>> It appears that 'unpack' has already executed for 'pydotplus-2.0.2'; skipping.
>>> Remove '/var/tmp/portage/dev-python/pydotplus-2.0.2/.unpacked' to force unpack.
>>> It appears that 'prepare' has already executed for 'pydotplus-2.0.2'; skipping.
>>> Remove '/var/tmp/portage/dev-python/pydotplus-2.0.2/.prepared' to force prepare.
>>> It appears that 'configure' has already executed for 'pydotplus-2.0.2'; skipping.
>>> Remove '/var/tmp/portage/dev-python/pydotplus-2.0.2/.configured' to force configure.
>>> It appears that 'compile' has already executed for 'pydotplus-2.0.2'; skipping.
>>> Remove '/var/tmp/portage/dev-python/pydotplus-2.0.2/.compiled' to force compile.
>>> Test phase: dev-python/pydotplus-2.0.2
 * python2_7: running distutils-r1_run_phase python_test

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
GLOB sdist-make: /var/tmp/portage/dev-python/pydotplus-2.0.2/work/pydotplus-2.0.2/setup.py
python create: /var/tmp/portage/dev-python/pydotplus-2.0.2/work/pydotplus-2.0.2/.tox/python
python inst: /var/tmp/portage/dev-python/pydotplus-2.0.2/work/pydotplus-2.0.2/.tox/dist/pydotplus-2.0.2.zip
python installed: appdirs==1.4.0,packaging==16.8,pydotplus==2.0.2,pyparsing==2.1.10,six==1.10.0
python runtests: PYTHONHASHSEED='1141240847'
___________________________________________________ summary ____________________________________________________
  python: commands succeeded
  congratulations :)
 * python3_4: running distutils-r1_run_phase python_test

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
GLOB sdist-make: /var/tmp/portage/dev-python/pydotplus-2.0.2/work/pydotplus-2.0.2/setup.py
python recreate: /var/tmp/portage/dev-python/pydotplus-2.0.2/work/pydotplus-2.0.2/.tox/python
python inst: /var/tmp/portage/dev-python/pydotplus-2.0.2/work/pydotplus-2.0.2/.tox/dist/pydotplus-2.0.2.zip
python installed: appdirs==1.4.0,packaging==16.8,pydotplus==2.0.2,pyparsing==2.1.10,six==1.10.0
python runtests: PYTHONHASHSEED='721560650'
___________________________________________________ summary ____________________________________________________
  python: commands succeeded
  congratulations :)
 * python2_7: running distutils-r1_run_phase _clean_egg_info
 * python3_4: running distutils-r1_run_phase _clean_egg_info
>>> Completed testing dev-python/pydotplus-2.0.2
```

**Comment by TheChymera** [comment]
@gktrk @SoapGentoo 

Hey guys, I think I'm calling all the pertinent tests, but the output still looks like no test is actually being run. Can you help me out?

**Comment by gktrk** [comment]
> Hey guys, I think I'm calling all the pertinent tests, but the output still looks like no test is actually being run. Can you help me out?

It's not your fault, the pypi tarball doesn't contain the `tox.init` file required by the tests

**Comment by TheChymera** [comment]
Can we get this in the tree then?

**Comment by gktrk** [comment]
I think this looks good enough for now. Let's comment out lines 18, 22-28, 35-38 because the tests aren't currently functional, file a bug on https://github.com/carlos-jenkins/pydotplus/issues and request the `tox.ini` file to be included in the pypi tarball for future releases.

We should also look into building and installing man pages and possibly html documentation. I'm currently not convinced that the man pages are not of any value and not sure how generated html files are compared to the built-in python `help()`

**Comment by gktrk** [comment]
One last thing, you should rebase all these commits into a single commit. Take a look at https://wiki.gentoo.org/wiki/Gentoo_Github#Rebasing_commits and let me know.

**Comment by TheChymera** [comment]
@gktrk @SoapGentoo - fixed, reported, re-tested. Works fine.

**Comment by gktrk** [comment]
Merged in https://github.com/gentoo/gentoo/compare/9ded54f...101f920. Thanks for all the hard work!

**Pull request closed without merging**